### PR TITLE
Profiles

### PR DESCRIPTION
**Pull request by strekm**
**Description**

Changes proposed in this pull request:

- profile support

**Related issue(s)**
see #159

### CODE REVIEW

**Review by Demonsthere — APPROVED** [review]
/meow

File: parallel-install/pkg/components/component.go [hunk 1]
```diff
@@ -42,10 +43,11 @@ type KymaComponent struct {
 //"chartDir" is a local filesystem directory with the component's chart.
 //
 //"overrides" is a function that returns overrides for the release.
-func NewComponent(name, namespace, chartDir string, overrides func() map[string]interface{}, helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
+func NewComponent(name, namespace, profile, chartDir string, overrides func() map[string]interface{}, helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
```
Comment (tobiscr) [inline parallel-install/pkg/components/component.go:46]: This is a quite long parameter list.. wouldn't it be better to use a param-struct to make it easier to read?

File: parallel-install/pkg/components/components.go [hunk 2]
```diff
@@ -65,15 +67,8 @@ func (p *ComponentsProvider) GetComponents() ([]KymaComponent, error) {
...
-		component := KymaComponent{
-			Name:            component.Name,
-			Namespace:       component.Namespace,
-			OverridesGetter: p.overridesProvider.OverridesGetterFunctionFor(component.Name),
-			ChartDir:        path.Join(p.resourcesPath, component.Name),
-			HelmClient:      helmClient,
-			Log:             p.log,
-		}
-		components = append(components, component)
+		cmp := NewComponent(component.Name, component.Namespace, p.profile, path.Join(p.resourcesPath, component.Name), p.overridesProvider.OverridesGetterFunctionFor(component.Name), helmClient, p.log)
```
Comment (tobiscr) [inline parallel-install/pkg/components/components.go:70]: Same as above.. maybe worth to switch to a param-struct instead passing so many parameter to a function call. this is pretty hard to read.

File: parallel-install/pkg/components/prerequisites.go [hunk 3]
```diff
@@ -50,14 +52,8 @@ func (p *PrerequisitesProvider) GetComponents() ([]KymaComponent, error) {
...
 
-		components = append(components, KymaComponent{
-			Name:            name,
-			Namespace:       namespace,
-			ChartDir:        path.Join(p.resourcesPath, name),
-			OverridesGetter: p.overridesProvider.OverridesGetterFunctionFor(name),
-			HelmClient:      helmClient,
-			Log:             p.log,
-		})
+		cmp := NewComponent(name, namespace, p.profile, path.Join(p.resourcesPath, name), p.overridesProvider.OverridesGetterFunctionFor(name), helmClient, p.log)
```
Comment (tobiscr) [inline parallel-install/pkg/components/prerequisites.go:55]: As above :)

File: parallel-install/pkg/components/component.go (hunk 1 above)
Comment (tobiscr) [reply parallel-install/pkg/components/component.go:46]: Example:

There are actually two options (at least I have in mind) to address long/complex parameter lists:

**Option 1. Param object**

        #Before
        func NewComponent(name, namespace, profile, chartDir string, overrides func() map[string]interface{},         helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
       ...
       }
        
        # With parameter object:
        struct NewComponentParams  struct {
           Name string
           Namespace string
           Profile string
           Chart string
           Overrides func() map[string]interface{}
           HelmClient helm.ClientInterface
           Log func(string, ...interface{}
        }
        func NewComponent(params NewComponentParams) {...}

**Option 2. Builder pattern**

        #Before
        func NewComponent(name, namespace, profile, chartDir string, overrides func() map[string]interface{}, helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
        ...
        }

        # a) Declarative way:
        compBuilder := ComponentBuilder{
          Name: "the name",
          Namespace: "theNamespace",
          EvaluationProfile: true,
          ChartName: "..."
          ...
        }
        comp := compBuilder.build()
        
       # b) Functional way
       compBuilder := ComponentBuilder{}
       comp := compBuilder.name("thename").
           namespace("thenamespace").
           useEvaluationProfile().
           chartNAme("chartanme").
           withHelmclient(helmclient).
           build()

File: parallel-install/README.md [hunk 4]
```diff
@@ -32,6 +32,7 @@ See all available configuration options for the `config.Config` type:
 | BackoffInitialIntervalSeconds | `int` | `1` | Initial interval used for exponential backoff retry policy. |
 | BackoffMaxElapsedTimeSeconds | `int` | `30` | Maximum time used for exponential backoff retry policy. |
 | Log | `func(format string, v ...interface{})` | `fmt.Printf` | Function used for logging. |
+| Profile | `string` | `evaluation` | Deployment profile: "evaluation", "production", ""  |
```
Comment (klaudiagrz) [inline parallel-install/README.md:35]:
```suggestion
| Profile | `string` | `evaluation` | Deployment profile. The possible values are: "evaluation", "production".  |
```

**Review by klaudiagrz — APPROVED** [review]

**Review by tobiscr — APPROVED** [review]

**Pull request merged**